### PR TITLE
Added segmented registration flow

### PR DESCRIPTION
**Pull request by youssef-attia**
### Summary 

Registration needed to be split into two steps with one being basic information and the next being auxiliary information. This PR splits registration into those two steps.

### Test Plan 

Standard Postman testing.

### CODE REVIEW

File: server/src/auth/auth.service.ts [hunk 1]
```diff
@@ -128,16 +129,17 @@ export class AuthService {
...
       user = await this.userService.register(
         idToken.email,
-        req.username,
-        req.year,
-        req.major,
         req.lat,
         req.long,
         authType,
         idToken.id,
+        req.username && req.year && req.major ? true : false,
```
Comment (erichuang27) [inline server/src/auth/auth.service.ts:139]: does this line up with the isRegistered field? Should have a comment to make it clear

**Review by erichuang27 — COMMENTED** [review]
Looks good other than that comment, up to you if you want to add

**Review by cgu2020 — COMMENTED** [review]
Check user.service. dtoForUserData requires your isRegistered field. Not sure if others functions require it but make sure it builds.

**Comment by neketka** [comment]
Nirbhay has discovered a simpler way of doing this without extra database work

**Pull request closed without merging**